vessel-runtime/src/test/java/com/textnow/android/vessel/VesselCacheUsageTest.kt

```diff
@@ -0,0 +1,295 @@
+package com.textnow.android.vessel
+
+/**
+ * MIT License
+ *
+ * Copyright (c) 2020 TextNow, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including witho`Fƒut limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import android.os.Build
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.Observer
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNull
+import kotlinx.coroutines.runBlocking
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.annotation.Config
+
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.take
+import kotlin.reflect.KClass
+
+import com.textnow.android.vessel.model.*
+import org.junit.Rule
+
+class TestCache : DefaultCache() {
+    val numberOfSetCalls = HashMap<String, Int>()
+
+    override fun <T : Any> set(key: String, value: T) {
+        numberOfSetCalls[key] = numberOfSetCalls.getOrDefault(key, 0) + 1
+        super.set(key, value)
+    }
+}
+
+data class Data1(val field: Int = 1)
```

Comment:
There are existing nested models that can be used for improved testing under src/test/java/com/textnow/android/vessel/model